exp/controllers/rosamachinepool_controller_test.go

```diff
@@ -60,3 +79,245 @@ func TestNodePoolToRosaMachinePoolSpec(t *testing.T) {
+				return nodePool, nil
+			}).Times(1)
+		}
+		expect(ocmMock.EXPECT())
+
+		c := fake.NewClientBuilder().WithScheme(scheme).WithObjects(rosaMachinePool, ownerCluster, ownerMachinePool, rosaControlPlane, secret).Build()
+		stsMock := mock_stsiface.NewMockSTSAPI(mockCtrl)
+		stsMock.EXPECT().GetCallerIdentity(gomock.Any()).Times(1)
+
+		r := ROSAMachinePoolReconciler{
+			Recorder:         recorder,
+			WatchFilterValue: "",
+			Endpoints:        []scope.ServiceEndpoint{},
+			Client:           c,
+			NewStsClient:     func(cloud.ScopeUsage, cloud.Session, logger.Wrapper, runtime.Object) stsiface.STSAPI { return stsMock },
+			NewOCMClient: func(ctx context.Context, rosaScope *scope.ROSAControlPlaneScope) (rosa.OCMClient, error) {
+				return ocmMock, nil
+			},
+		}
+
+		req := ctrl.Request{}
+		req.NamespacedName = types.NamespacedName{Name: "rosa-machinepool", Namespace: ns.Name}
+
+		result, err := r.Reconcile(ctx, req)
+
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(result).To(Equal(ctrl.Result{}))
+	})
+
+	t.Run("Reconcile delete", func(t *testing.T) {
+		setup(t)
+		defer teardown()
+
+		deleteTime := metav1.NewTime(time.Now().Add(5 * time.Second))
+		rosaMachinePool.ObjectMeta.Finalizers = []string{"finalizer-rosa"}
+		rosaMachinePool.ObjectMeta.DeletionTimestamp = &deleteTime
+
+		ocmMock := mocks.NewMockOCMClient(mockCtrl)
+		expect := func(m *mocks.MockOCMClientMockRecorder) {
+			m.GetNodePool(gomock.Any(), gomock.Any()).DoAndReturn(func(clusterId string, nodePoolID string) (*cmv1.NodePool, bool, error) {
+				nodePoolBuilder := nodePoolBuilder(rosaMachinePool.Spec, ownerMachinePool.Spec)
+				nodePool, err := nodePoolBuilder.ID("node-pool-1").Build()
+				g.Expect(err).To(BeNil())
+				return nodePool, true, nil
+			}).Times(1)
+			m.DeleteNodePool("rosa-control-plane1", "node-pool-1").DoAndReturn(func(clusterId string, nodePoolID string) error {
+				return nil
+			}).Times(1)
+		}
+		expect(ocmMock.EXPECT())
+
+		client := fake.NewClientBuilder().WithScheme(scheme).WithObjects(rosaMachinePool, ownerCluster, ownerMachinePool, rosaControlPlane, secret).Build()
+		stsMock := mock_stsiface.NewMockSTSAPI(mockCtrl)
+		stsMock.EXPECT().GetCallerIdentity(gomock.Any()).Times(1)
+
+		r := ROSAMachinePoolReconciler{
+			Recorder:         recorder,
+			WatchFilterValue: "",
+			Endpoints:        []scope.ServiceEndpoint{},
+			Client:           client,
+			NewStsClient:     func(cloud.ScopeUsage, cloud.Session, logger.Wrapper, runtime.Object) stsiface.STSAPI { return stsMock },
+			NewOCMClient: func(ctx context.Context, rosaScope *scope.ROSAControlPlaneScope) (rosa.OCMClient, error) {
+				return ocmMock, nil
+			},
+		}
+
+		req := ctrl.Request{}
+		req.NamespacedName = types.NamespacedName{Name: "rosa-machinepool", Namespace: ns.Name}
+
+		result, err := r.Reconcile(ctx, req)
+		g.Expect(err).ToNot(HaveOccurred())
+		g.Expect(result).To(Equal(ctrl.Result{}))
```

Comment:
You need to validate if the RosaMachinePool deleted.

Reply:
I am testing that `DeleteNodePool`. It seems that `reconcileDelete` doesn't modify the `ROSAMachinePool` object directly.